### PR TITLE
apib 1.2.1

### PR DESCRIPTION
**Pull request by gbrail**
A few changes to the dependencies, and now builds using cmake.

- [X ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/apib.rb [hunk 1]
```diff
@@ -14,28 +13,21 @@ class Apib < Formula
...
-    inreplace "configure" do |s|
-      s.gsub! "/usr/include/apr-1.0", "#{Formula["apr"].opt_libexec}/include/apr-1"
-      s.gsub! "/usr/include/apr-1", "#{Formula["apr"].opt_libexec}/include/apr-1"
-    end
-    ENV.append "LDFLAGS", "-L#{Formula["apr-util"].opt_libexec}/lib"
-    ENV.append "LDFLAGS", "-L#{Formula["apr"].opt_libexec}/lib"
-    ENV.append "CFLAGS", "-I#{Formula["apr"].opt_libexec}/include/apr-1"
-    ENV.append "CFLAGS", "-I#{Formula["apr-util"].opt_libexec}/include/apr-1"
+    mkdir "release"
+    system "cmake", ".", "-DCMAKE_BUILD_TYPE=Release"
```
Comment (Bo98) [inline Formula/apib.rb:27]: Looks good. One change I'd make:

```suggestion
    system "cmake", ".", *std_cmake_args
```

File: Formula/apib.rb [hunk 2]
```diff
@@ -14,28 +13,21 @@ class Apib < Formula
...
-    # https://github.com/apigee/apib/issues/11
-    inreplace "configure" do |s|
-      s.gsub! "/usr/include/apr-1.0", "#{Formula["apr"].opt_libexec}/include/apr-1"
-      s.gsub! "/usr/include/apr-1", "#{Formula["apr"].opt_libexec}/include/apr-1"
-    end
-    ENV.append "LDFLAGS", "-L#{Formula["apr-util"].opt_libexec}/lib"
-    ENV.append "LDFLAGS", "-L#{Formula["apr"].opt_libexec}/lib"
-    ENV.append "CFLAGS", "-I#{Formula["apr"].opt_libexec}/include/apr-1"
-    ENV.append "CFLAGS", "-I#{Formula["apr-util"].opt_libexec}/include/apr-1"
+    mkdir "release"
```
Comment (Bo98) [inline Formula/apib.rb:26]: Is this used? You seem to create a directory but don't use it.

File: Formula/apib.rb [hunk 3]
```diff
@@ -14,28 +13,21 @@ class Apib < Formula
...
+  depends_on "libev"
   depends_on "openssl@1.1"
 
   def install
-    # Fix detection of libcrypto for OpenSSL 1.1
-    inreplace "configure", "CRYPTO_num_locks", "EVP_sha1"
+    ENV.append "EXTRA_INCLUDE_DIR", "${prefix}/include"
+    ENV.append "EXTRA_LIB_DIR", "${prefix}/lib"
+    ENV.append "OPENSSL_INCLUDE_DIR", "#{Formula["openssl@1.1"]}/include"
+    ENV.append "OPENSSL_LIB_DIR", "#{Formula["openssl@1.1"]}/lib"
```
Comment (Bo98) [inline Formula/apib.rb:24]: This doesn't do as you expect - it outputs "openssl@1.1/include". Given the build passes, it's possible it isn't needed.

File: Formula/apib.rb [hunk 4]
```diff
@@ -14,28 +13,21 @@ class Apib < Formula
...
-  depends_on "apr-util"
+  depends_on "cmake" => :build
+  depends_on "libev"
   depends_on "openssl@1.1"
 
   def install
-    # Fix detection of libcrypto for OpenSSL 1.1
-    inreplace "configure", "CRYPTO_num_locks", "EVP_sha1"
+    ENV.append "EXTRA_INCLUDE_DIR", "${prefix}/include"
+    ENV.append "EXTRA_LIB_DIR", "${prefix}/lib"
```
Comment (Bo98) [inline Formula/apib.rb:22]: `prefix` in this context actually is set to "/usr/local/Cellar/apib/1.2.1`. Again, I suspect it is not needed here - Homebrew automatically adds the include and lib directories of the dependencies.

**Comment by gbrail** [comment]
Thanks -- using the changes you suggested the whole thing builds with much less pain and suffering! I just pushed another commit that cleans up the formula.

**Review by Bo98 — APPROVED** [review]
Nice and clean - thanks! I'll merge this in a moment.

**Pull request closed without merging**